### PR TITLE
Stage Hidden Costs of Infra Management blog

### PR DESCRIPTION
**Pull request by shughes26**

### CODE REVIEW

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+date: 2024-12-09T09:00:00-08:00
+meta_desc: "nfrastructure as Code (IaC) has revolutionized how cloud resources are managed, allowing for more efficient, scalable, and repeatable deployments."
+meta_image: meta.png
+authors:
+    - aaron-kao
+tags:
+    - infrastructure-as-code
+---
+
+Infrastructure as Code (IaC) has revolutionized how cloud resources are managed, allowing for more efficient, scalable, and repeatable deployments. We designed Pulumi IaC to let you program cloud infrastructure using familiar programming languages like TypeScript, JavaScript, Python, Go, .NET, Java, and YAML. This approach not only simplifies the process but also integrates seamlessly with existing development tools and ecosystems (e.g., IDEs, standard unit test frameworks, integration test). You can define infrastructure with code, often in just one line, for serverless, Kubernetes, AI/ML, databases, and more. You can also preview changes before deploying unlike many other IaC solutions. Pulumi IaC is fully open source with a [public roadmap](https://github.com/orgs/pulumi/projects/44/). We value working with the community to shape the product through feedback and contributions.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:12]:
```suggestion
[Infrastructure as Code (IaC)](/what-is/what-is-infrastructure-as-code/) has revolutionized how cloud resources are managed, allowing for more efficient, scalable, and repeatable deployments. We designed [Pulumi IaC](/product/infrastructure-as-code/) to let you program cloud infrastructure using familiar programming languages like TypeScript, JavaScript, Python, Go, .NET, Java, and YAML. This approach not only simplifies the process but also integrates seamlessly with existing development tools and ecosystems (e.g., IDEs, standard unit test frameworks, integration test). You can define infrastructure with code, often in just one line, for serverless, Kubernetes, AI/ML, databases, and more. You can also preview changes before deploying unlike many other IaC solutions. Pulumi IaC is fully open source with a [public roadmap](https://github.com/orgs/pulumi/projects/44/). We value working with the community to shape the product through feedback and contributions.
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+When it comes to scaling, the cost of onboarding new teams or users onto a DIY system can be high due to the need for custom documentation, training, and support. Security is another critical area where the costs can escalate, with the need to comply with numerous IT security standards, manage access control, and ensure robust secrets management.
+
+We built Pulumi Cloud as a fully managed service to address these concerns by providing a platform that automates deployments, ensures compliance, and offers comprehensive security features out-of-the-box. You’ll reduce your operational burden, minimize your risk of outages, and make scaling and onboarding easier, ultimately leading to cost savings and increased productivity. By choosing Pulumi Cloud, your organization can focus on innovation and growth rather than the intricacies of IaC backend management.You get to optimize your infrastructure without the overhead of running your own DIY solution.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:106]:
```suggestion
We built Pulumi Cloud as a fully managed service to address these concerns by providing a platform that automates deployments, ensures compliance, and offers comprehensive security features out-of-the-box. You’ll reduce your operational burden, minimize your risk of outages, and make scaling and onboarding easier, ultimately leading to cost savings and increased productivity. By choosing Pulumi Cloud, your organization can focus on innovation and growth rather than the intricacies of IaC backend management. You get to optimize your infrastructure without the overhead of running your own DIY solution.
```

Can you please check for typos everywhere else?

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+## Next Steps
+
+Each unique variable in your environment - cloud providers, languages, team structure, business objective, etc. - makes realizing the full benefits of Pulumi Cloud different for each organization.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:110]: these should be em dashes

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+1. Strategic Leadership Engagement: Address key business challenges like security posture, cost optimization, and developer productivity.
+2. Architectural Review: Evaluate and refine platform engineering projects to enhance efficiency and scalability.
+3. Hands-On Workshops: Equip your engineering teams with practical Pulumi skills through interactive technical sessions.
+
+This program will help strengthen executive buy-in for your infrastructure initiatives while showcasing Pulumi's value to your peer engineering teams.
+
+If this sounds valuable, let’s discuss tailoring the agenda to fit your organization’s priorities.
+
+Register now, or contact your Pulumi sales team.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:124]: We need a link here

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+*"Terraform relies on HCL and lacks support for concepts like classes, objects and inheritance. An equivalent deployment would take more lines of code while yielding IaC that is less reusable."*  
+
+[Learn more.](/case-studies/unity/)
+{{% /notes %}}
+
+### Security and Compliance
+
+IaC backends are mission-critical components that require robust security measures to protect the backend service, cloud assets, and applications running on the infrastructure.
+
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:56]: Can you put the grey image holders in here so we can see how everything looks

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+## What is Pulumi Cloud?
+
+Infrastructure as Code tools like Pulumi require systems for coordinating deployments, which include concurrency control, state management, and security. For the purposes of this post, let's refer to these systems as IaC backends. There are several options for managing these backends: you can either handle them yourself, which we'll call DIY, or you can leverage Pulumi Cloud, available as a SaaS or self-hosted solution. With Pulumi Cloud, you gain access to a comprehensive infrastructure management platform designed to handle everything running in the cloud. This platform automates your IaC deployments, centralizes secrets management and orchestration to manage secrets sprawl effectively, and employs AI to oversee infrastructure assets and ensure compliance. On the other hand, with a DIY approach, you have the flexibility to build everything Pulumi Cloud offers or opt for a more minimalistic setup tailored to your organization's specific needs. Both options are well-supported by Pulumi, but there are distinct advantages to choosing Pulumi Cloud over the DIY method, which we'll explore further.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:16]:
```suggestion
Infrastructure as Code tools like Pulumi require systems for coordinating deployments, which include concurrency control, state management, and security. For the purposes of this post, let's refer to these systems as IaC backends. There are several options for managing these backends: you can either handle them yourself, which we'll call DIY, or you can leverage [Pulumi Cloud](/product/pulumi-cloud/), available as a SaaS or self-hosted solution. With Pulumi Cloud, you gain access to a comprehensive infrastructure management platform designed to handle everything running in the cloud. This platform automates your IaC deployments, centralizes secrets management and orchestration to manage secrets sprawl effectively, and employs AI to oversee infrastructure assets and ensure compliance. On the other hand, with a DIY approach, you have the flexibility to build everything Pulumi Cloud offers or opt for a more minimalistic setup tailored to your organization's specific needs. Both options are well-supported by Pulumi, but there are distinct advantages to choosing Pulumi Cloud over the DIY method, which we'll explore further.
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+Pulumi Cloud further enhances security with Pulumi ESC, a centralized secrets management and orchestration service. This service allows for pulling and syncing secrets from various stores like HashiCorp Vault, AWS Secrets Manager, Azure Key Vault, GCP Secret Manager, and 1Password, making it easy to adopt dynamic, short-lived secrets on demand. Pulumi ESC leverages the same identity, RBAC, Teams, SAML/SCIM, OIDC, and scoped access tokens used for Pulumi IaC, ensuring that secrets management complies with enterprise security policies. Every access or change to secrets or configuration values is logged for auditing, providing a secure and auditable environment for you and your team to access secrets via CLI, API, Kubernetes operator, the Pulumi Cloud UI, and in-code with TypeScript/JavaScript, Python, and Go SDKs. With a DIY backend, you have to manage integration and access controls with individual secrets stores, with potential secrets sprawl and operational overhead.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:68]:
```suggestion
Pulumi Cloud further enhances security with [Pulumi ESC](/product/secrets-management/), a centralized secrets management and orchestration service. This service allows for pulling and syncing secrets from various stores like HashiCorp Vault, AWS Secrets Manager, Azure Key Vault, GCP Secret Manager, and 1Password, making it easy to adopt dynamic, short-lived secrets on demand. Pulumi ESC leverages the same identity, RBAC, Teams, SAML/SCIM, OIDC, and scoped access tokens used for Pulumi IaC, ensuring that secrets management complies with enterprise security policies. Every access or change to secrets or configuration values is logged for auditing, providing a secure and auditable environment for you and your team to access secrets via CLI, API, Kubernetes operator, the Pulumi Cloud UI, and in-code with TypeScript/JavaScript, Python, and Go SDKs. With a DIY backend, you have to manage integration and access controls with individual secrets stores, with potential secrets sprawl and operational overhead.
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
+---
+title: "The Hidden Costs of Infrastructure Management"
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:2]:
```suggestion
title: "The Hidden Costs of Infrastructure as Code"
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+Managing your infrastructure by IaC demands considerable time and energy. Pulumi Cloud provides Pulumi Insights, offering search, analytics, and AI-driven insights over your infrastructure. If you're using a DIY backend, you'll need to build your own search clusters and grep systems. With Insights, you can instantly search for critical information - like finding MySQL databases on end-of-life versions across all your cloud assets. You can also have a conversation with Pulumi Copilot about your infrastructure in plain English and get
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:38]:
```suggestion
Managing your infrastructure by IaC demands considerable time and energy. Pulumi Cloud provides [Pulumi Insights](/product/pulumi-insights/), offering search, analytics, and AI-driven insights over your infrastructure. If you're using a DIY backend, you'll need to build your own search clusters and grep systems. With Insights, you can instantly search for critical information - like finding MySQL databases on end-of-life versions across all your cloud assets. You can also have a conversation with [Pulumi Copilot](/product/copilot/) about your infrastructure in plain English and get
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+Audit logs play a vital role in tracking user activity within an organization, recording what actions were taken, when, and by whom. Pulumi's audit logs are immutable, capturing the UNIX timestamp of events, the user who invoked the action, the event itself, and the source IP of the call. This level of auditing ensures that the activities of organization members are attributable, enhancing security and compliance.
+
+At the cloud asset level, you and your team need to prevent infrastructure deployments that violate policies related to security, reliability, cost, or compliance. Pulumi Cloud, through Pulumi CrossGuard, provides out-of-the-box support for common security and compliance policies, ensuring PCIDSS, ISO27001, SOC2, and CIS compliance for cloud applications and infrastructure. Such organization-wide enforcement and specific policy support are not readily available with DIY backends.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:64]:
```suggestion
At the cloud asset level, you and your team need to prevent infrastructure deployments that violate policies related to security, reliability, cost, or compliance. Pulumi Cloud, through [Pulumi CrossGuard](/crossguard/), provides out-of-the-box support for common security and compliance policies, ensuring PCIDSS, ISO27001, SOC2, and CIS compliance for cloud applications and infrastructure. Such organization-wide enforcement and specific policy support are not readily available with DIY backends.
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
+---
+title: "The Hidden Costs of Infrastructure Management"
+date: 2024-12-09T09:00:00-08:00
+meta_desc: "nfrastructure as Code (IaC) has revolutionized how cloud resources are managed, allowing for more efficient, scalable, and repeatable deployments."
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:4]:
```suggestion
meta_desc: "IaC backends have hidden costs. Learn how Pulumi Cloud reduces operational burden, minimizes risk of outages, and make scaling easier."
```

**Review by aaronkao — COMMENTED** [review]
Some edits

**Comment by aaronkao** [comment]
Adding this here for the social:  https://www.perplexity.ai/search/come-up-with-a-linkedin-post-a-uadgUHT8Q9KW4jB1oa0LLQ

**Review by aaronkao — COMMENTED** [review]
Some more comments

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
+---
+title: "The Hidden Costs of Infrastructure as Code"
+date: 2024-12-09T09:00:00-08:00
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:3]:
```suggestion
date: 2024-12-10T09:00:00-08:00
```

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+## Costs and Complexities of DIY Backends
+
+Building and maintaining your own Infrastructure as Code backend involves a significant engineering and operational commitment. On the engineering front, you face security and compliance challenges, as well as the need to develop features that ensure enterprise readiness. Operationally, the backend must be highly reliable and available, and it should facilitate easy onboarding for the rest of the organization when the time comes. Beyond these immediate concerns, there are also hidden personnel costs associated with running your own backend. Let's break down some of those hidden costs..
```
Comment (thtmnisamnstr) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:20]:
```suggestion
Building and maintaining your own Infrastructure as Code backend involves a significant engineering and operational commitment. On the engineering front, you face security and compliance challenges, as well as the need to develop features that ensure enterprise readiness. Operationally, the backend must be highly available and reliable, and it should facilitate easy onboarding for the rest of the organization when the time comes. Beyond these immediate concerns, there are also hidden personnel costs associated with running your own backend. Let's break down some of those hidden costs..
```

**Review by thtmnisamnstr — APPROVED** [review]

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+answers to operational questions that would take hours to piece together from other sources.
+
+![Pulumi Copilot example prompt](copilot.png)
+
+With this total visibility, Pulumi Insights helps you reduce waste by showing you usage information and identifying potentially stale infrastructure, leading to substantial cost savings.
+
+When you build and operate your own IaC backend, you'll need to dedicate one full-time engineer for every team of ten - that's nearly 10% of your engineering expense. Pulumi Cloud, as a fully managed service, frees up your team's time from the operational and maintenance burdens of managing an IaC backend.
```
[the remark this answers is not in the document]
Comment (aaronkao) [reply content/blog/hidden-costs-of-infrastructure-management/index.md:45]: lets leave it out and assume that

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+
+1. **Strategic Leadership Engagement:** Address key business challenges like security posture, cost optimization, and developer productivity.
+2. **Architectural Review:** Evaluate and refine platform engineering projects to enhance efficiency and scalability.
+3. **Hands-On Workshops:** Equip your engineering teams with practical Pulumi skills through interactive technical sessions.
+
+This program will help strengthen executive buy-in for your infrastructure initiatives while showcasing Pulumi's value to your peer engineering teams.
+
+If this sounds valuable, let’s discuss tailoring the agenda to fit your organization’s priorities.
+
+Register now or contact your Pulumi sales team.
```
[the remark this answers is not in the document]
Comment (aaronkao) [reply content/blog/hidden-costs-of-infrastructure-management/index.md:132]: https://info.pulumi.com/pulumi-platform-engineering-immersion-days - yeah it didn't get created until an hour ago

File: content/blog/hidden-costs-of-infrastructure-management/index.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+
+When you’re scaling your organization, you’ll need an IaC backend that makes team onboarding efficient. DIY backends often involve ad-hoc onboarding with bespoke identity solutions, requiring custom documentation and training to troubleshoot unique problems. This can slow down onboarding and reduce end-user productivity. Additionally, DIY backends place the internal support burden on your team, requiring them to assist users in navigating the system.
+
+In contrast, Pulumi Cloud offers standardized, well-documented capabilities and integrates seamlessly with identity systems, leading to efficient onboarding and increased productivity for new employees. Pulumi Cloud provides [12x5 or 24x7 support](https://support.pulumi.com/hc/en-us), ensuring your organization receives immediate assistance with any architectural, cloud-related, or Pulumi best-practices issues. Furthermore, Pulumi Cloud includes built-in role-based access control (RBAC), simplifying the onboarding and scaling of new users.
```
Comment (aaronkao) [inline content/blog/hidden-costs-of-infrastructure-management/index.md:92]:
```suggestion
In contrast, Pulumi Cloud offers standardized, well-documented capabilities and integrates seamlessly with identity systems, leading to efficient onboarding and increased productivity for new employees. Pulumi Cloud provides [12x5 or 24x7 support](https://support.pulumi.com/hc/en-us), ensuring your organization receives immediate assistance with any architectural, cloud-related, or Pulumi best-practices issues. Furthermore, Pulumi Cloud includes built-in RBAC, simplifying the onboarding and scaling of new users.
```
